imagersite/imager_images/views.py

```diff
@@ -1,3 +1,61 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from .models import Album, Photo
 
-# Create your views here.
+
+def photo_view(request, photo_id):
+    photo = Photo.objects.filter(id=photo_id).first()
+    username = request.user.get_username()
+    context = {}
+    if photo.user.username == username or photo.published == 'PUBLIC':
+        context['photo'] = photo
+
+    return render(request, 'imager_images/photo.html', context)
+
+
+def photo_gallery_view(request):
+    username = request.user.get_username()
+    gallery = Photo.objects.filter(
+                user__username=username).filter(published='PUBLIC')
```

Comment:
The gallery displayed is supposed to be _all_ the public photos uploaded to the app.

Reply:
done